ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+      command: update-ca-trust extract
+      become: true
+    
+    - name: Create container to serve the registry
+      shell: |
+        [[ $( podman ps -a --format '{{podman_format_name | quote }}' --filter name={{ pod_name_registry }} ) ]] \
+          && {
+               podman container stop {{ pod_name_registry }}
+               podman container rm {{ pod_name_registry }}
+             }
+    
+        podman create \
+          --name {{ pod_name_registry }} \
+          -p {{ registry_port_container }}:{{ registry_port }} \
+          -v {{ registry_dir_data }}:/var/lib/registry:z \
+          -v {{ registry_dir_auth }}:/auth:z \
+          -e "REGISTRY_AUTH=htpasswd" \
+          -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry" \
+          -e "REGISTRY_HTTP_SECRET=ALongRandomSecretForRegistry" \
+          -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+          -v {{ registry_dir_cert }}:/certs:z \
+          -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+          -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key docker.io/library/registry:2
+      become: true
+    
+    - name: Start container
+      command: podman start {{ pod_name_registry }}
+      become: true
+    
+    - name: Mirror remote registry to local
+      command:
+        oc adm release mirror
+          -a "{{ homedir }}/pullsecret.txt"
+          --from={{ release_image | quote }}
+          --to-release-image={{ local_registry | quote }}/{{ local_repo | quote }}:{{ release_version | quote }}
+          --to={{ local_registry | quote }}/{{ local_repo | quote }}
+    
+    - name: Read in the contents of domain.crt
+      slurp:
+        src: "{{ registry_dir_cert }}/domain.crt"
+      register: domain_cert_b64
+    
+    - name: Set trustbundle fact to contents of domain.crt
+      set_fact:
+        trustbundle: "{{ domain_cert_b64.content | string | b64decode }}"
+    
+    - name: Create mirrorred registry information
+      set_fact:
+        install_config_appends: "{{ lookup('template', 'install-config-appends.j2'
+    ) }}"
+    
+    - name: Create {{ install_config_appends_file }}
+      copy:
+        content: "{{ install_config_appends }}"
+        dest: "{{ homedir }}/{{ install_config_appends_file }}"
+    
+    - name: Create {{ install_config_appends_file }} on localhost
+      copy:
+        content: "{{ install_config_appends }}"
+        dest: "~/{{ install_config_appends_file }}"
+      delegate_to: localhost
+    
+    - name: Information
+      debug:
+        msg:
+        - "To use this mirrored registry to deploy you must do the following"
+        - "Add the authentication from either "
+        - "    {{ homedir }}/{{ registry_auth_file }} on {{ inventory_hostname }}"
+        - "    or ~/{{ registry_auth_file }} on this server to your pull secret."
+        - ""
+        - "Append the contents of either of the "
+        - "    {{ homedir }}/{{ install_config_appends_file }} on {{ inventory_hostname }} "
```

Comment:
anywhere you used homedir, use the {{ ansible_env.HOME }}

Reply:
done will include in next commit.